Automated Smoketest Build: failed
Number of Error(s): 4
/mnt/disk1/home/vamosax/smoketest/smoketest-7844/HPCC-Platform/roxie/ccd/ccdlistener.cpp:674: error: ‘cpuset’ was not declared in this scope
make[2]: *** [roxie/ccd/CMakeFiles/ccd.dir/ccdlistener.cpp.o] Error 1
make[1]: *** [roxie/ccd/CMakeFiles/ccd.dir/all] Error 2
make: *** [all] Error 2

--- a/roxie/ccd/ccdlistener.cpp
+++ b/roxie/ccd/ccdlistener.cpp
@@ -670,7 +670,17 @@ public:
         }
         else
         {
+#if __GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 6)
             cpuCores = CPU_COUNT(&cpuMask);
+#else
+            cpuCores = 0;
+            unsigned setSize = CPU_SETSIZE;
+            while (setSize--)
+            {
+                if (CPU_ISSET(setSize, &cpuMask))
+                    ++cpuCores;
+            }
+#endif /* GLIBC */
             if (traceLevel)
                 traceAffinity(&cpuMask);
         }

